From a5b718f5beca3b64e471408fcd2c8a218da9bb21 Mon Sep 17 00:00:00 2001
From: Yixun Lan <dlan@gentoo.org>
Date: Thu, 28 Aug 2025 22:56:23 +0000
Subject: [PATCH 12/33] hack: sunxi: ddr: a733: skip DDR init, retrieve size
 only

Due to no DDR code available, let's skip the initialization then,
and only return a fixed DDR size, which hardcoded to 2GB, it should
be enough, and work for most boards.

Signed-off-by: Yixun Lan <dlan@gentoo.org>
---
 arch/arm/mach-sunxi/Kconfig            | 3 +++
 arch/arm/mach-sunxi/Makefile           | 1 +
 arch/arm/mach-sunxi/dram_sun60i_a733.c | 6 ++++++
 3 files changed, 10 insertions(+)
 create mode 100644 arch/arm/mach-sunxi/dram_sun60i_a733.c

diff --git a/arch/arm/mach-sunxi/Kconfig b/arch/arm/mach-sunxi/Kconfig
index b7ac828e16d..0aa25362751 100644
--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -741,6 +741,7 @@ config DRAM_CLK
 	default 744 if MACH_SUN50I_H6
 	default 720 if MACH_SUN50I_H616 || MACH_SUN50I_A133
 	default 1200 if MACH_SUN55I_A523
+	default 0
 	---help---
 	Set the dram clock speed, valid range 240 - 480 (prior to sun9i),
 	must be a multiple of 24. For the sun9i (A80), the tested values
@@ -767,6 +768,7 @@ config DRAM_ZQ
 	default 3881979 if MACH_SUNXI_H3_H5 || MACH_SUN8I_R40 || MACH_SUN50I_H6
 	default 4145117 if MACH_SUN9I
 	default 3881915 if MACH_SUN50I
+	default 0
 	---help---
 	Set the dram zq value.
 
@@ -779,6 +781,7 @@ config DRAM_ODT_EN
 	default y if MACH_SUN8I_R40
 	default y if MACH_SUN50I
 	default y if MACH_SUN50I_H6
+	default 0
 	---help---
 	Select this to enable dram odt (on die termination).
 
diff --git a/arch/arm/mach-sunxi/Makefile b/arch/arm/mach-sunxi/Makefile
index 9c79b55abf3..685ad8c253e 100644
--- a/arch/arm/mach-sunxi/Makefile
+++ b/arch/arm/mach-sunxi/Makefile
@@ -49,4 +49,5 @@ obj-$(CONFIG_DRAM_SUN50I_A133)	+= dram_sun50i_a133.o
 obj-$(CONFIG_DRAM_SUN50I_A133)	+= dram_timings/
 obj-$(CONFIG_MACH_SUN55I_A523)	+= dram_sun55i_a523.o dram_dw_helpers.o
 obj-$(CONFIG_DRAM_SUN55I_A523)	+= dram_timings/
+obj-$(CONFIG_MACH_SUN60I_A733)	+= dram_sun60i_a733.o
 endif
diff --git a/arch/arm/mach-sunxi/dram_sun60i_a733.c b/arch/arm/mach-sunxi/dram_sun60i_a733.c
new file mode 100644
index 00000000000..cd4079ccff4
--- /dev/null
+++ b/arch/arm/mach-sunxi/dram_sun60i_a733.c
@@ -0,0 +1,6 @@
+#include <linux/sizes.h>
+
+unsigned long sunxi_dram_init(void)
+{
+	return SZ_2G;
+}
-- 
2.53.0

